gaptrain/active.py

```diff
@@ -142,13 +143,119 @@ def get_active_config(config, gap, temp, e_thresh, max_time_fs,
     curr_time_fs += md_time_fs
 
     # If the prediction is within the threshold then call this function again
-    return get_active_config(config, gap, temp, e_thresh, max_time_fs,
-                             curr_time_fs=curr_time_fs,
-                             method_name=method_name,
-                             n_calls=n_calls+1)
+    return get_active_config_true(config, gap, temp, e_thresh, max_time_fs,
+                                  curr_time_fs=curr_time_fs,
+                                  ref_method_name=ref_method_name,
+                                  n_calls=n_calls+1)
 
 
-def get_active_configs(config, gap, method_name,
+def get_active_config_qbc(config, gap, temp, std_e_thresh, max_time_fs):
+    """
+    Generate an 'active' configuration i.e. a configuration to be added to the
+    training set by active learning using a query-by-committee model where
+    above some standard deviation in the prediction between different models
+    exceeds a threshold (std_e_thresh)
+
+    ------------------------------------------------------------------------
+    :param config: (gt.Configuration)
+    :param gap: (gt.GAPEnsemble)
+    :param temp: (float)
+    :param std_e_thresh: (float) eV
+    :param max_time_fs: (float)
+    :return: (gt.Configuration)
+    """
+    n_iters, curr_time = 0, 0.0
+
+    while curr_time < max_time_fs:
+
+        gap_traj = gt.md.run_gapmd(config,
+                                   gap=gap.gaps[0],
+                                   temp=float(temp),
+                                   dt=0.5,
+                                   interval=4,
+                                   fs=2 + n_iters**3,
+                                   n_cores=1)
+
+        for frame in gap_traj[::max(1, len(gap_traj)//10)]:
+            pred_es = []
+            # Calculated predicted energies in serial for all the gaps
+            for single_gap in gap.gaps:
+                frame.run_gap(gap=single_gap, n_cores=1)
+                pred_es.append(frame.energy)
+
+            # and return the frame if the standard deviation in the predictions
+            # is larger than a threshold
+            std_e = np.std(np.array(pred_es))
+            if std_e > std_e_thresh:
+                return frame
+
+            logger.info(f'σ(t={curr_time:.1f}) = {std_e:.6f}')
+
+        n_iters += 1
+        curr_time += 2 + n_iters**3
```

Comment:
Why do you pass fs=2 + n_iters**3 but then in curr_time you add fs but as it is written explicitly instead of just passing 'fs'? Do you use fs?